### PR TITLE
Feature/bubble sort app

### PR DESCRIPTION
**Pull request by tnyandoro**
# A bubble_sort nd bubble_sort_by
> Initialised the length of the array
>set the  loop to sorted false
>Set the method of the array length to if the first array is greater than the preceding one swap then compare.
> Loop the list again and test the block condition, swap, and loop again.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      end
+    end
+
+    break unless swapped
+  end
+
+  arr
+end
+
+p bubble_sort([4, 3, 78, 2, 0, 2])
```
Comment (maelfosso) [inline bubble_sort.rb]: Not really needed.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        arr[j], arr[j + 1] = arr[j + 1], arr[j]
+        swapped = true
+      end
+    end
+    break unless swapped
+  end
+  arr
+end
+
+puts bubble_sort_by(%w[hello hi hey]) { |a, b| a.length - b.length }
```
Comment (maelfosso) [inline bubble_sort.rb]: Not needed.

File: README.md [hunk 3]
```diff
@@ -1 +1,57 @@
...
+[@tnyandoro](https://repl.it/)
+
+# Getting started
+
+To get a local copy of this repo, please run the following command in your terminal
+
+
+$ cd<folder>
+
+$ git clone [](https://github.com/tnyandoro/bubble-sort/tree/feature/bubble-sort_app)
```
Comment (maelfosso) [inline README.md]: The result of this line is `git clone ` without the repository URL.
Change it with 
```
git clone https://github.com/tnyandoro/bubble-sort/tree/feature/bubble-sort_app
```

Complete your README with how to run your code

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for the well done job so far :blush: 

1. There are some linters' errors. Kindly fix them.

2. Read the comments left below.

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

**Review by Rhelli — CHANGES_REQUESTED** [review]
# Changes Required 🛑 ✅

Hi @tnyandoro, excellent work on the project, I've tested it and all tests are passing. You are however not passing the required linter tests, so please amend the linter errors and re-submit for another code review to get this project approved.

If you need any help you can @ me here on Github or you can reach me in Slack.

All the best,

💂🏽 Rory

**Comment by tnyandoro** [comment]
Fixed Lint Errors

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hello @tnyandoro 

📝 I see you tried to avoid the linter warnings by modifying the rules, please do not do that, remember:

- [ ] Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

👉 I see the main problem is with the linter warning of the crlf issue, that mainly happens when you use a code editor in windows, to avoid that, you can: 

a) configure your editor

b) use an online editor, you can go to this virtual environment (just append gitpod.io/ to your repo URL) -> https://gitpod.io/github.com/tnyandoro/bubble-sort/tree/feature/bubble-sort_app and retype your code in a new file and delete the old file that has the troublesome configuration

Please, use the original rubocop file https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml

### ♻️ Status: Your project requires some additional changes ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me in slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate if can leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😛 

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

**_Note:_** All the **[OPTIONAL]** are **not required** in order to approve your project but are personal suggestions to improve your coding experience

**Review by javitocor — APPROVED** [review]
# STATUS APPROVED 🥇 🌟 

Hello team! 
You have met all the requirements and the project has been approved! 👏 
Congratulations! 💯 👍 
Remember to fill up the completion form 📑  and move into the next project ⏩ 

Good luck with it! 😺

**Pull request merged**